### PR TITLE
Sticky contacts

### LINKED ISSUE #615: Investigate sliding stability
The stability of a sliding box has been somewhat improved by recent changes (better contact point retention, limiting breaks along tangent), but it still visually 'wobbles' in many cases.

### PR DESCRIPTION
**Pull request by McCallisterRomer**
resolves #615

Building upon previous work to preserve tangentially separated contacts. That feature works well reduce jitter on sliding boxes, but it can be problematic in other cases. Reworking this a bit so that the manifold can make slightly more informed decisions with respect to discarding/preserving. We will only extend contact persistence if:
- `EnableStickyContacts==true`
- The collision event is between a dynamic and static object.
- The collision event doesn't involve spheres, capsules, or concave geometry.
- The separation tangent to the contact normal doesn't exceed `StickyContactMaxDistance`.

### CODE REVIEW

File: source/engine/physics/PhysicsConstants.h [hunk 1]
```diff
@@ -16,19 +16,21 @@ constexpr bool EnableJointWarmstarting  = true;  // Start joint resolution from
...
-constexpr float PenetrationSlop = 0.005f;                    // How far objects are allowed to penetrate
-constexpr bool EnableBaumgarteStabilization = true;          // Enable velocity-based position correction
-constexpr float BaumgarteFactor = 0.3f;                      // Bias factor for baumgarte stabilization [0, 1]
-constexpr bool EnableDirectPositionCorrection = false;       // Enable translation-based position correction
-constexpr float PositionCorrectionFactor = 0.2f;             // Percentage of mtv to use for direct position correction
-constexpr float MaxLinearPositionCorrection = 0.2f;          // Maximum translation distance for direct position correction
-constexpr bool EnableRestitutionSlop = true;                 // Enable zeroing of restitution below a threshold
-constexpr float RestitutionSlop = 0.001f;                    // Velocity threshold for applying restitution
+constexpr size_t EpaMaxIterations = 32u;             // Number of iterations aftwerwhich to give up trying to find contact points
```
Comment (jaremieromer) [inline source/engine/physics/PhysicsConstants.h]: *after which

File: source/engine/physics/PhysicsConstants.h [hunk 2]
```diff
@@ -16,19 +16,21 @@ constexpr bool EnableJointWarmstarting  = true;  // Start joint resolution from
...
+constexpr float EpaTolerance = 0.00001f;             // Error threshold for finding contact points
```
Comment (jaremieromer) [inline source/engine/physics/PhysicsConstants.h]: Will you add [0,1] (or whatever the range is) to all of these like some of the others?

File: source/engine/physics/PhysicsConstants.h [hunk 3]
```diff
@@ -16,19 +16,21 @@ constexpr bool EnableJointWarmstarting  = true;  // Start joint resolution from
...
+constexpr float ContactBreakDistance = -0.01f;       // How aggressively contact points are removed/retained
+constexpr bool EnableStickyContacts = true;          // Minimize how many contacts points are discarded due to tangential separation (only for dynamic box/hull vs static box/hull)
+constexpr float StickyContactBreakDistance = -0.5f;  // How aggressively sticky contact points are removed/retained
+constexpr float StickyContactMaxDistance = 1.0f;     // Always discard contacts sticky contacts beyond this threshold
+
+constexpr float PenetrationSlop = 0.005f;               // How far objects are allowed to penetrate
+constexpr bool EnableBaumgarteStabilization = true;     // Enable velocity-based position correction
+constexpr float BaumgarteFactor = 0.3f;                 // Bias factor for baumgarte stabilization [0, 1]
+constexpr bool EnableDirectPositionCorrection = false;  // Enable translation-based position correction
+constexpr float PositionCorrectionFactor = 0.2f;        // Percentage of mtv to use for direct position correction
```
Comment (jaremieromer) [inline source/engine/physics/PhysicsConstants.h]: What is MTV?

File: source/engine/physics/collision/narrow_phase/NarrowPhase.cpp [hunk 4]
```diff
@@ -154,16 +136,9 @@ void NarrowPhase::NotifyEvents()
...
             {
-                /** This isn't expected to be detected here, but just in case. */
-                NC_LOG_CONTACTS("Clear stale manifold [unexpected code path]");
-                const auto e1 = cur->Event().first;
-                const auto e2 = cur->Event().second;
-                if(auto* logic = TryGetCollisionLogic(e1)) logic->NotifyCollisionExit(e2, m_registry);
-                if(auto* logic = TryGetCollisionLogic(e2)) logic->NotifyCollisionExit(e1, m_registry);
-
-                m_manifoldCache.Remove(e1, e2);
+                // No new collision this frame, but manfifold found contacts still within break tolerances
```
Comment (jaremieromer) [inline source/engine/physics/collision/narrow_phase/NarrowPhase.cpp]: *manifold

**Review by jaremieromer — APPROVED** [review]

File: source/engine/physics/PhysicsConstants.h (hunk 3 above)
Comment (McCallisterRomer) [reply source/engine/physics/PhysicsConstants.h]: minimum translation vector - will change

**Pull request merged**